tests/aws/services/cloudformation/resource_providers/ec2/test_ec2.py

```diff
@@ -56,6 +56,29 @@ def test_deploy_prefix_list(deploy_cfn_template, aws_client, snapshot):
     snapshot.add_transformer(snapshot.transform.key_value("PrefixListId"))
 
 
+@markers.aws.validated
+def test_deploy_security_group_with_tags(deploy_cfn_template, aws_client, snapshot):
+    """Create security group in default VPC with tags."""
+    stack = deploy_cfn_template(
+        template_path=os.path.join(
+            os.path.dirname(__file__), "../../../../templates/ec2_security_group_with_tags.yml"
+        )
+    )
+
+    snapshot.add_transformer(snapshot.transform.key_value("GroupId"))
+    snapshot.add_transformer(snapshot.transform.key_value("GroupName"))
+    snapshot.add_transformer(snapshot.transform.key_value("VpcId"))
+    snapshot.add_transformer(SortingTransformer("Tags", lambda tag: tag["Key"]))
+    response = aws_client.ec2.describe_security_groups(GroupIds=[stack.outputs["SecurityGroupId"]])
+    security_group = response["SecurityGroups"][0]
+
+    security_group["Tags"] = [
+        tag for tag in security_group["Tags"] if not tag["Key"].startswith("aws:cloudformation")
+    ]
```

Comment:
Nice use of transformers.

The marker `@markers.snapshot.skip_verify` enables the skipping of certain attributes from validation using json paths, but it may not be applicable in this case. It's great that you filtered the tags that the resource provider, Another option could be for the resource provider to add them (even though it may be beyond the current scope of the issue).